src/bmp.c

```diff
@@ -4,14 +4,16 @@
 
 #include "bmp.h"
 
+char new_filename[100];

```

Comment:
You missed moving this into the write_bmp function.

Reply:
done